cpp/locality/LinkCell.cc

```diff
@@ -269,93 +269,91 @@ void LinkCell::compute(const box::Box& box,
         });
     }
 
-void LinkCell::computeCellNeighbors()
+const std::vector<unsigned int>& LinkCell::computeCellNeighbors(unsigned int cur_cell)
     {
-    // clear the list
-    m_cell_neighbors.clear();
-    m_cell_neighbors.resize(getNumCells());
-
-    // for each cell
-    for (unsigned int k = 0; k < m_cell_index.getD(); k++)
-        for (unsigned int j = 0; j < m_cell_index.getH(); j++)
-            for (unsigned int i = 0; i < m_cell_index.getW(); i++)
-                {
-                // clear the list
-                unsigned int cur_cell = m_cell_index(i,j,k);
-                m_cell_neighbors[cur_cell].clear();
-
-                // loop over the neighbor cells
-                int starti, startj, startk;
-                int endi, endj, endk;
-                if (m_celldim.x < 3)
-                    {
-                    starti = (int)i;
-                    }
-                else
-                    {
-                    starti = (int)i - 1;
-                    }
-                if (m_celldim.y < 3)
-                    {
-                    startj = (int)j;
-                    }
-                else
-                    {
-                    startj = (int)j - 1;
-                    }
-                if (m_celldim.z < 3)
-                    {
-                    startk = (int)k;
-                    }
-                else
-                    {
-                    startk = (int)k - 1;
-                    }
+    std::vector<unsigned int> neighbor_cells;
+    vec3<unsigned int> l_idx = m_cell_index(cur_cell);
+    unsigned int i = l_idx.x;
```

Comment:
I'd like to cast to `int` here, instead of casting `i`, `j`, `k` to `int` later.